src/public/js/pipe-web-monetization.js

```diff
@@ -1,32 +1,76 @@
-(function( $ ) {
-	'use strict';
-
-	/**
-	 * All of the code for your public-facing JavaScript source
-	 * should reside in this file.
-	 *
-	 * Note: It has been assumed you will write jQuery code here, so the
-	 * $ function reference has been prepared for usage within the scope
-	 * of this function.
-	 *
-	 * This enables you to define handlers, for when the DOM is ready:
-	 *
-	 * $(function() {
-	 *
-	 * });
-	 *
-	 * When the window is loaded:
-	 *
-	 * $( window ).on('load', function() {
-	 *
-	 * });
-	 *
-	 * ...and/or other possibilities.
-	 *
-	 * Ideally, it is not considered best practice to attach more than a
-	 * single DOM-ready or window-load handler for a particular page.
-	 * Although scripts in the WordPress core, Plugins and Themes may be
-	 * practising this, we should strive to set a better example in our own work.
-	 */
-
-})( jQuery );
+jQuery(function ($) {
+  var count = 0;
+  var totalAmount = 0;
+  
+  if (!ajax_variables.logged_in) {
+    $.ajax({
+      url: ajax_variables.ajax_url,
+      type: 'post',
+      data: { random: Math.random() },
+      success: function (response) {
+        $("head").append(response);
+        simulateMonetization()
+        // document.head.appendChild(response)
+      }
+    })
+  }
+  if (document.monetization) {
+    document.monetization.addEventListener('monetizationstart', () => {
+      console.log("start")
+      //   document.getElementById('exclusive').classList.remove('hidden')
+    })
+    document.monetization.addEventListener('monetizationprogress', event => {
+      console.log("progress")
+      count++;
+      totalAmount += Number(event.detail.amount);
+      console.log(Number(event.detail.amount))
+    })
```

Comment:
Great! We should include the batcher here too in order to prepare for performing the request.